### PR TITLE
Moved flush to after_call

### PR DESCRIPTION
**Pull request by dsblank**
## Details

This PR adds @track(flush=True) so that you don't have to do it manually after the function has been called.

Before:

```python
from opik import track, flush_tracker

@track()
def streaming_function(input):
    messages = [{"role": "user", "content": input}]
    response = framework.completion(
        model="gpt-3.5-turbo",
        messages=messages,
    )
    return response
streaming_function("Why?")
flush_tracker()
```

After:

```python
from opik import track

@track(flush=True)
def streaming_function(input):
    messages = [{"role": "user", "content": input}]
    response = framework.completion(
        model="gpt-3.5-turbo",
        messages=messages,
    )
    return response
streaming_function("Why?")
```

## Issues

None, except for adding some convenience.

## Testing

No tests added.

## Documentation

Docstring for flush added.

### CODE REVIEW

**Comment by jverre** [comment]
This look good to me, thanks :) Let's wait for @japdubengsub to review and approve before we merge

@dsblank Do you think we should update the docs ? Could be good to add a paragraph in the log_traces page

**Review by japdubengsub — APPROVED** [review]
Unfortunately, I wasn't part of the initial discussion on this task since I joined the project a bit later, so I might be slightly out of context.
Everything looks okay!
My only concern is that this might negatively impact batching mode. However, since both this mode and batching are disabled by default, there shouldn't be any significant issues. :)

**Comment by dsblank** [comment]
@andrescrz, I can't find a test of `tracker.flush_tracker` and I don't have enough information to test it myself. Can you point me to something that I could use as a base to copy?

**Comment by andrescrz** [comment]
> @andrescrz, I can't find a test of `tracker.flush_tracker` and I don't have enough information to test it myself. Can you point me to something that I could use as a base to copy?

As discussed offline, this was tested manually prior to merge, so it's fine to go from my end. We can add a test if any issue or at a later stage.

**Pull request merged**
